// @bugsnag/plugin-angular/dist/esm/index.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { ErrorHandler, Injectable } from "@angular/core";
import { Bugsnag } from "@bugsnag/js";
let BugsnagErrorHandler = class BugsnagErrorHandler extends ErrorHandler {
    constructor(bugsnagClient) {
        super();
        this.bugsnagClient = bugsnagClient;
    }
    handleError(error) {
        const handledState = {
            severity: "error",
            severityReason: { type: "unhandledException" },
            unhandled: true,
        };
        const report = new this.bugsnagClient.BugsnagReport(error.name, error.message, this.bugsnagClient.BugsnagReport.getStacktrace(error), handledState, error);
        if (error.ngDebugContext) {
            report.updateMetaData("angular", {
                component: error.ngDebugContext.component,
                context: error.ngDebugContext.context,
            });
        }
        this.bugsnagClient.notify(report);
        ErrorHandler.prototype.handleError.call(this, error);
    }
};
BugsnagErrorHandler = __decorate([
    Injectable(),
    __metadata("design:paramtypes", [Bugsnag.Client])
], BugsnagErrorHandler);
export { BugsnagErrorHandler };
const ɵ0 = (client) => {
    return new BugsnagErrorHandler(client);
};
const plugin = {
    init: ɵ0,
    name: "Angular",
};
export default plugin;
export { ɵ0 };
